docker/Dockerfile.build

```diff
@@ -1,56 +1,88 @@
-# Builder image
-FROM rust:buster as builder
+# This steps implement cargo-chef for docker layer caching
+# We are using four stages:
+# - chef: installs cargo-chef
+# - planner: computes the recipe file
+# - builder: caches our dependencies and builds the binary
+# - runtime: is our runtime environment
+FROM rust:bullseye as chef
+RUN cargo install cargo-chef
+WORKDIR /app
+
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare  --recipe-path recipe.json
+
+FROM chef AS builder
+COPY --from=planner /app/recipe.json recipe.json
 
 RUN apt-get update && \
-	apt-get install -y --no-install-recommends \
-	make cmake g++ gcc llvm libclang-dev clang ca-certificates
+    apt-get install -y --no-install-recommends \
+    make cmake g++ gcc llvm libclang-dev clang ca-certificates && \
+    rm -Rf /var/lib/apt/lists/* /tmp/*
 
-RUN mkdir /zebra
-WORKDIR /zebra
+# Optimize builds. In particular, regenerate-stateful-test-disks.yml was reaching the
+# GitHub Actions time limit (6 hours), so we needed to make it faster.
+ENV RUSTFLAGS -O
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --features enable-sentry --recipe-path recipe.json
 
+# Build zebra
 ARG SHORT_SHA
 ENV SHORT_SHA $SHORT_SHA
+ARG RUST_BACKTRACE
+ENV RUST_BACKTRACE ${RUST_BACKTRACE:-full}
+# Skip test on debian based OS by default as compilation fails
+ARG ZEBRA_SKIP_NETWORK_TESTS
+ENV ZEBRA_SKIP_NETWORK_TESTS ${ZEBRA_SKIP_NETWORK_TESTS:-1}
 
-ENV RUST_BACKTRACE full
-ENV CARGO_HOME /zebra/.cargo/
+ENV CARGO_HOME /app/.cargo/
 
 RUN rustc -V; cargo -V; rustup -V
 
 COPY . .
-
-RUN cd zebrad/; cargo build --release --features enable-sentry
+# Build and then compile tests
+# Compile, but don't run tests (--no-run). Add verbosity and colors
+RUN cargo build --release --features enable-sentry --bin zebrad
+RUN cargo test --release --features enable-sentry --workspace --no-run
 
```

Comment:
I don't think we want to enable Sentry (error reporting service) for tests

Reply:
If we don't supply the same `cargo` features to each command, we'll re-compile `zebrad` just for the tests.

Is there a way of turning off sentry using an environmental variable?